lib/ansible/modules/cloud/oneandone/oneandone.py

```diff
@@ -0,0 +1,793 @@
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: oneandone
+short_description: create, destroy, start, stop, and reboot a 1&1 Host machine.
+description:
+     - create, destroy, update, start, stop, and reboot a 1&1 Host machine.
+       When the machine is created it can optionally wait for it to be
+       'running' before returning.
+       This module has a dependency on 1and1 >= 1.0
+version_added: "2.1"
+options:
+  state:
+    description:
+      - Define a machine's state to create, remove, start or stop it.
+    required: false
+    default: 'present'
+    choices: [ "present", "absent", "running", "stopped" ]
+  auth_token:
+    description:
+      - Authenticating API token provided by 1&1. Overrides the
+        ONEANDONE_AUTH_TOKEN environement variable.
+    required: true
+  datacenter:
+    description:
+      - The datacenter location.
+    required: false
+    default: US
+    choices: [ "US", "ES", "DE", "GB" ]
+  hostname:
+    description:
+      - The hostname or ID of the machine. Only used when state is 'present'.
+    required: true when state is 'present', false otherwise.
+  description:
+    description:
+      - The description of the machine.
+    required: false
+  appliance:
+    description:
+      - The operating system name or ID for the machine.
+    required: true for 'present' state, false otherwise
+  fixed_instance_size:
+    description:
+      - The instance size name or ID of the machine.
+    required: true for 'present' state, false otherwise
+    choices: [ "S", "M", "L", "XL", "XXL", "3XL", "4XL", "5XL" ]
+  vcore:
+    description:
+      - The total number of processors.
+    required: true with cores_per_processor, ram, and hdds
+  cores_per_processor:
+    description:
+      - The number of cores per processor.
+    required: true with vcore, ram, and hdds
+  ram:
+    description:
+      - The amount of RAM memory.
+    required: true with vcore, cores_per_processor, and hdds
+  hdds:
+    description:
+      - A list of hard disks with nested "size" and "is_main" properties.
+    required: true with vcore, cores_per_processor, and ram
+  private_network:
+    description:
+      - The private network name or ID.
+    required: false
+  firewall_policy:
+    description:
+      - The firewall policy name or ID.
+    required: false
+  load_balancer:
+    description:
+      - The load balancer name or ID.
+    required: false
+  monitoring_policy:
+    description:
+      - The monitoring policy name or ID.
+    required: false
+  instance_ids:
+    description:
+      - List of machine IDs or hostnames.
+    required: false for 'running' state, true otherwise
+  count:
+    description:
+      - The number of machines to create.
+    required: false
+    default: 1
+  ssh_key:
+    description:
+      - User's public SSH key.
```

Comment:
Is this a path to a key, or the contents of the key. Please update `description:` and example to be clearer.